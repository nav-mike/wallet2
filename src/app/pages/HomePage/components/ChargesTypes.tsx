import React from 'react';
import { VictoryPie } from 'victory';

type ChargeType = {
  x: string;
  y: number;
};

type ChargesTypesData = {
  charges: ChargeType[];
};

export function ChargesTypes(props: ChargesTypesData) {
  return <VictoryPie data={props.charges} />;
}
