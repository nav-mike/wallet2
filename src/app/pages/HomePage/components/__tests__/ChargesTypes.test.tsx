import * as React from 'react';
import { render } from '@testing-library/react';
import { ChargesTypes } from '../ChargesTypes';

describe('<ChargesTypes />', () => {
  const charges = [
    { x: 'Food', y: 23 },
    { x: 'Transport', y: 55.2 },
  ];

  it('should render and match the snapshot', () => {
    const types = render(<ChargesTypes charges={charges} />);
    expect(types.container.firstChild).toMatchSnapshot();
  });
});
