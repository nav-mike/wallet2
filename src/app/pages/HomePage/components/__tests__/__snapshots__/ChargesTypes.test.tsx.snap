// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChargesTypes /> should render and match the snapshot 1`] = `
<div
  class="VictoryContainer"
  style="height: 100%; width: 100%; pointer-events: none; position: relative;"
>
  <svg
    height="400"
    role="img"
    style="pointer-events: all; width: 100%; height: 100%;"
    viewBox="0 0 400 400"
    width="400"
  >
    <g>
      <path
        d="M9.184850993605149e-15,-150A150,150,0,0,1,144.27384647592288,41.04944851081241L0,0Z"
        role="presentation"
        shape-rendering="auto"
        style="fill: #252525; padding: 10px; stroke: transparent; stroke-width: 1;"
        transform="translate(200, 200)"
      />
      <path
        d="M144.27384647592288,41.04944851081241A150,150,0,1,1,-2.7554552980815446e-14,-150L0,0Z"
        role="presentation"
        shape-rendering="auto"
        style="fill: #525252; padding: 10px; stroke: transparent; stroke-width: 1;"
        transform="translate(200, 200)"
      />
      <text
        direction="inherit"
        dx="0"
        id="pie-labels-0"
        x="336"
        y="102.97"
      >
        <tspan
          dx="0"
          dy="0"
          style="padding: 20px; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; fill: #252525; stroke: transparent;"
          text-anchor="start"
          x="336"
        >
          Food
        </tspan>
      </text>
      <text
        direction="inherit"
        dx="0"
        id="pie-labels-1"
        x="64"
        y="306.97"
      >
        <tspan
          dx="0"
          dy="0"
          style="padding: 20px; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; fill: #252525; stroke: transparent;"
          text-anchor="end"
          x="64"
        >
          Transport
        </tspan>
      </text>
    </g>
  </svg>
  <div
    style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
  >
    <svg
      height="400"
      style="overflow: visible; width: 100%; height: 100%;"
      viewBox="0 0 400 400"
      width="400"
    />
  </div>
</div>
`;
